server/src/test/java/org/candlepin/policy/AutobindRulesTest.java

```diff
@@ -3031,13 +3045,13 @@ public void testSelectBestPoolsAddonMatchShouldOverrideAllOtherSyspurposeMismatc
         // --- No satisfied syspurpose attributes on the consumer ---
 
         Product prod1 = createSysPurposeProduct(null, "mismatched_role", "my_addon",
-            "mismatched_sla", "mismatched_usage");
+            "mismatched_sla", "mismatched_usage", null);
```

Comment:
The consumer should have a service_type set, and this should be "random_service_type" (that mismatches with the consumer's)

Reply:
Done.